Assets/HoloToolkit/Input/Scripts/GazeManager.cs

```diff
@@ -151,11 +167,53 @@ private void UpdateRaycast()
             {
                 if (oldFocusedObject != null)
                 {
+                    IInteractable oldInteractable;
+
+                    if (!tapCache.TryGetValue(oldFocusedObject, out oldInteractable))
+                    {
+                        oldInteractable = oldFocusedObject.GetComponent<IInteractable>();
+                        tapCache.Add(oldFocusedObject, oldInteractable);
+                    }
+
+                    if (oldInteractable != null)
+                    {
+                        oldInteractable.OnGazeExit();
+                    }
+
+                    if (OnGazeExit != null)
+                    {
+                        OnGazeExit(oldFocusedObject);
+                    }
+
+                    // Obsolete! Please subscribe to OnGazeExit event.
                     oldFocusedObject.SendMessage("OnGazeLeave", SendMessageOptions.DontRequireReceiver);
+                    SendMessage("OnGazeLeave", SendMessageOptions.DontRequireReceiver);
+                    // End Obsolete
                 }
                 if (FocusedObject != null)
                 {
+                    IInteractable newInteractable;
```

Comment:
yes, but it would you wouldn't have to look in the cache for the oldInteractable a few lines earlier and during the events from gesture recognizer.  You also won't have to look it up if the focused object doesn't change frame to frame.